src/olcs/abstractsynchronizer.js

```diff
@@ -131,6 +131,26 @@ olcs.AbstractSynchronizer.prototype.addLayers_ = function(root) {
       }
     } else {
       cesiumObjects = this.createSingleLayerCounterparts(olLayerWithParents);
+      if (!cesiumObjects) {
+        // keep an eye on the layers that once failed to be added (might work when the layer is updated)
+        // for example when a source is set after the layer is added to the map
+        const layerId = olLayerId;
+        const layerWithParents = olLayerWithParents;
+        const onLayerChange = (e) => {
+          const cesiumObjs = this.createSingleLayerCounterparts(layerWithParents);
+          if (cesiumObjs) {
+            // unsubscribe event listener
+            layerWithParents.layer.un('change', onLayerChange, this);
+            this.layerMap[layerId] = cesiumObjs;
+            this.olLayerListenKeys[layerId].push(ol.events.listen(layerWithParents.layer, 'change:zIndex', this.orderLayers, this));
+            cesiumObjs.forEach(function(cesiumObject) {
+              this.addCesiumObject(cesiumObject);
+            }, this);
```

Comment:
This code block is the same as a few lines below. Could you move it into a private method: `addCesiumObjects_`?